File: CS50/CS50x/sentimental-readability/readability.py
# TODO

def main():
    text = promptText()
    counter = countText(text)
    grade = round(5.88 * counter[2] / counter[0] - 29.6 * counter[1] / counter[0] - 15.8)
    if (grade < 1):
        print("Before Grade 1")
    elif (grade >= 16):
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def promptText():
    return input("Text: ")


def countText(text):
    counter = [0 for i in range(3)]
    for c in text:
        if (c == ' '):
            counter[0] += 1
        elif (isEnding(c)):
            counter[1] += 1
        elif (isLetter(c)):
            counter[2] += 1
    counter[0] += 1
    return counter


def isLetter(c):
    if (c >= 'A' and c <= 'Z'):
        return True
    if (c >= 'a' and c <= 'z'):
        return True
    return False


def isEnding(c):
    if (c == '.' or c == '!' or c == '?'):
        return True
    return False


if __name__ == "__main__":
    main()

File: CS50/CS50x/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <stdbool.h>
#include <string.h>
#include <math.h>

string promptText(void);
void countText(string text, int properties[]);
bool isLetter(char c);
bool isWhitespace(char c);
bool isSentenceEnd(char c);
double computeIndex(int properties[]);
void findGradeLevel(double index);

int main(void)
{
    string text = promptText();
    int properties[] = {0, 0, 0};
    countText(text, properties);
    findGradeLevel(computeIndex(properties));
}

string promptText(void)
{
    return get_string("Text: ");

}

void countText(string text, int properties[])
{
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isLetter(text[i]))
        {
            properties[0]++;
        }
        else if (isWhitespace(text[i]))
        {
            properties[1]++;
        }
        else if (isSentenceEnd(text[i]))
        {
            properties[2]++;
        }
    }
    properties[1]++;
}

bool isLetter(char c)
{
    return (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z');
}

bool isWhitespace(char c)
{
    return c == ' ';
}

bool isSentenceEnd(char c)
{
    return c == '.' || c == '?' || c == '!';
}

double computeIndex(int properties[])
{
    return 5.88 * properties[0] / properties[1] - 29.6 * properties[2] / properties[1] - 15.8;
}

void findGradeLevel(double index)
{
    int grade = (int) round(index);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}